crates/sui-core/src/authority/authority_store_tables.rs

```diff
@@ -566,10 +564,16 @@ fn effects_table_default_config() -> DBOptions {
 }
 
 fn events_table_default_config() -> DBOptions {
-    default_db_options()
-        .optimize_for_write_throughput()
-        .optimize_for_large_values_no_scan(4 << 10)
-        .optimize_for_read(read_size_from_env(ENV_VAR_EVENTS_BLOCK_CACHE_SIZE).unwrap_or(1024))
+    DBOptions {
+        options: default_db_options()
+            .optimize_for_write_throughput()
+            .optimize_for_large_values_no_scan(4 << 10)
+            .optimize_for_read(read_size_from_env(ENV_VAR_EVENTS_BLOCK_CACHE_SIZE).unwrap_or(1024))
+            .options,
+        rw_options: ReadWriteOptions {
+            apply_range_deletions: true,
```

Comment:
actually, I'd like to keep `apply_range_deletions=false` for the events cf too due to potential performance implications. Currently there's one call site outside of the pruner that performs range deletion for [events](https://github.com/MystenLabs/sui/blob/cd6bbe0682fa7c115aca5633b78e172573835794/crates/sui-core/src/authority/authority_store.rs#L1321).  If we set `apply_range_deletions=false` for the events cf there will be a short period of time when transaction got reverted but its events are still queryable. At first glance, it shouldn't cause any issues on a validator node. cc @mystenmark